cdns/cloudflare/update-dns.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+DIR="$(dirname "$0")"
+source "$DIR/config.sh"
+
+DNS_SERVER=8.8.8.8
+
+echo "CHALLENGE_DOMAIN: ${CHALLENGE_DOMAIN}"
+echo "CHALLENGE_VALUE: ${CERTBOT_VALIDATION}"
+echo "DNS_SERVER: ${DNS_SERVER}"
+echo "ZONE: ${CLOUDFLARE_ZONE}"
+
+ADD_RECORD_RESULT=$(curl -X POST "https://api.cloudflare.com/client/v4/zones/${CLOUDFLARE_ZONE}/dns_records" \
+     -H "X-Auth-Email: ${CLOUDFLARE_EMAIL}" \
+     -H "X-Auth-Key: ${CLOUDFLARE_KEY}" \
+     -H "Content-Type: application/json" \
+     --data "{\"type\":\"TXT\",\"name\":\"${CHALLENGE_DOMAIN}\",\"content\":\"${CERTBOT_VALIDATION}\", \"ttl\": 120}" -s | jq -r "[.success, .errors[].message] | @csv")
+
+echo "Add challange record result: ${ADD_RECORD_RESULT}"
+
+
+ADD_DNS_RESULT=$(curl -X POST "https://api.cloudflare.com/client/v4/zones/${CLOUDFLARE_ZONE}/dns_records" \
```

Comment:
What happens if the record already exists?

Reply:
By the way, this curl command can be in a function and then call the function each time.